### PR TITLE
JAVACLIENT-149: Add Nuxeo client version in User-Agent

### PR DESCRIPTION
**Pull request by kevinleturc**

### CODE REVIEW

**Comment by pierre-gautier** [comment]
👍

**Comment by pierre-gautier** [comment]
👍

**Comment by funshodavid** [comment]
:+1:

File: nuxeo-java-client/src/main/java/org/nuxeo/client/NuxeoClient.java [hunk 1]
```diff
@@ -92,6 +97,19 @@ protected void addOkHttpInterceptor(Interceptor interceptor) {
         buildRetrofit();
     }
 
+    protected String computeUserAgent() {
+        String nuxeoPart = " NuxeoJavaClient/";
+        try (InputStream inputStream = getClass().getResourceAsStream("/config.properties")) {
```
Comment (efge) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/NuxeoClient.java]: Could we have a more specific filename for this? The client will be a library used by various applications and we don't want to pollute the root of the classpath...

File: nuxeo-java-client/src/main/java/org/nuxeo/client/NuxeoClient.java [hunk 2]
```diff
@@ -92,6 +97,19 @@ protected void addOkHttpInterceptor(Interceptor interceptor) {
...
+            Properties properties = new Properties();
+            properties.load(inputStream);
+            String nuxeoVersion = properties.getProperty("nuxeo.java.client.version");
+            nuxeoPart += nuxeoVersion;
+        } catch (IOException e) {
+            nuxeoPart += "Unknown";
```
Comment (efge) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/NuxeoClient.java]: Maybe also log the error?

**Pull request merged**